### DISCUSSION TITLE
core-frontend.unhandledException | TypeError: Cannot read properties of undefined (reading 'x')

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#5343] [Q&A]
I recently started getting these errors on iModels being viewed from the iTwin Hub.  These are I models we have been creating from our own transformation service.  I have tried on iTwin.js version 3.6.1 and 3.7.0. There hasn't been any drastic changes in our code for creating iTwins and not sure where we should start to troubleshoot since we are not getting any errors on creation.  And no, reloading the page doesn't work.  

Also, after clicking on the ok message, the Tree View renders properly and lists all the model objects and properties properly. I think it has to do with the tile generation.

Any help would be most appreciative.  

![image](https://user-images.githubusercontent.com/51784707/229597555-a27cd0df-2a43-448d-a2db-cb1ae83972e4.png)

![image](https://user-images.githubusercontent.com/51784707/229597739-7a996ee1-b795-4521-b2c3-5b1defc8ce07.png)

**Comment by markschlosseratbentley** [comment]
The following PR was backported + included in the 3.7.1 release: https://github.com/iTwin/itwinjs-core/pull/5328
Your problem looks a bit different, but somewhat similar to what we observed in that issue.
I would recommend trying with 3.7.1 once that is feasible and letting us know if this still happens (it very well may).
If so, could you provide more details?
- Does it only happen with a specific set of iModels? If so, can a simple iModel be provided that demonstrates the issue?
- Can it be reproduced in a simpler environment? (A sandbox, for example)

**Reply by spowell-216** [reply]
I am trying 3.7.1 and should have answer in the morning. 
It seems to be happening with the iModels that we create from scratch.  The iModels that are generated from the connectors seems to be rendering just fine. 
I have tried opening in one of our iTwin viewer environments that is running with "@itwin/web-viewer-react": "^3.0.8" and I am getting the same error.
![image](https://user-images.githubusercontent.com/51784707/229629576-73a99df9-d710-4816-afdd-c359f2c777a2.png)



I think I know what my be contributing to the problem, I am trying to calculate the camera range using the following ECSQL query:

```select min(si.minX) as minX, min(si.minY) as minY, min(minZ) as minZ, max(si.maxX) as maxX, max(si.maxY) as maxY, max(si.maxZ) as maxZ from bis.SpatialIndex si```

and i am getting nulls on the Min x,y,z
![image](https://user-images.githubusercontent.com/51784707/229632965-86e96bf4-6f48-40ed-82ae-562c40c350ce.png)

funny thing is I can't filter them out. 
![image](https://user-images.githubusercontent.com/51784707/229633516-55268bc7-851e-472f-91be-159c833cbe6f.png)

do you by chance know of a better ECSQL query to get the bounding box of all the geometry for use as the camera range?

**Reply by markschlosseratbentley** [reply]
Hi @spowell-216 Yes, definitely having a bad camera specified in the iModel could cause an error like you reported above. As far as the correct ECSQL query, @ColinKerr do you know?

**Reply by spowell-216** [reply]
@markschlosseratbentley I have successfully created an iModel again. I made two changes overnight, 1. using 3.7.1 and 2. fixed my camera range query to: 
```
SELECT min(si.minX) as minX, min(si.minY) as minY, min(minZ) as minZ, max(si.maxX) as maxX, max(si.maxY) as maxY, max(si.maxZ) as maxZ
FROM bis.SpatialIndex si join bis.SpatialElement se on si.ECInstanceId = se.ECInstanceId 
WHERE se.Origin.x <> 0
```
![image](https://user-images.githubusercontent.com/51784707/229856900-f3c1aafa-5e19-47d7-8587-138b038d7295.png)

I am pretty sure my problem was related to the cameraRange query, you can mark this closed.  Thanks.
